src/argilla/client/feedback/dataset/remote/dataset.py

```diff
@@ -591,6 +601,34 @@ def add_metadata_property(
 
         return ArgillaMixin._parse_to_remote_metadata_property(metadata_property=metadata_property, client=self._client)
 
+    def vector_settings_by_name(self, name: str) -> RemoteVectorSettings:
+        # TODO: Maybe make sense to have a query param in api.list_vector_settings to filter by name
+        for vector_settings in self.vector_settings:
+            if vector_settings.name == name:
+                return vector_settings
+
+        raise KeyError(f"Vector settings with name {name!r} does not exist in the dataset.")
```

Comment:
If we choose to go for not raising an error when the vector_settings is not found then maybe we should do something similar here.

Reply:
I give you here the same argument as before